src/toast/toast.spec.ts

```diff
@@ -1,21 +1,21 @@
 import {Component} from '@angular/core';
-import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {async, ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
 
-import {createGenericTestComponent} from '../test/common';
+import {createGenericTestComponent, isBrowserVisible} from '../test/common';
 import {NgbToastModule} from './toast.module';
 
+import {NgbConfig} from '../ngb-config';
+import {NgbConfigAnimation} from '../test/ngb-config-animation';
+
 const createTestComponent = (html: string) =>
     createGenericTestComponent(html, TestComponent) as ComponentFixture<TestComponent>;
 
-const getElementWithSelector = (fixture: ComponentFixture<TestComponent>, className) =>
-    fixture.nativeElement.querySelector(className);
+const getElementWithSelector = (element: HTMLElement, className) => element.querySelector(className);
 
-const getToastElement = (fixture: ComponentFixture<TestComponent>): Element =>
-    getElementWithSelector(fixture, 'ngb-toast');
-const getToastHeaderElement = (fixture: ComponentFixture<TestComponent>): Element =>
-    getElementWithSelector(fixture, 'ngb-toast .toast-header');
-const getToastBodyElement = (fixture: ComponentFixture<TestComponent>): Element =>
-    getElementWithSelector(fixture, 'ngb-toast .toast-body');
+const getToastElement = (element: HTMLElement): Element => getElementWithSelector(element, 'ngb-toast');
+const getToastHeaderElement = (element: HTMLElement): Element =>
+    getElementWithSelector(element, 'ngb-toast .toast-header');
+const getToastBodyElement = (element: HTMLElement): Element => getElementWithSelector(element, 'ngb-toast .toast-body');
```

Comment:
Why? Because of that change, you had to put `fixture.nativeElement` everywhere in the file...

Reply:
It's because I have this error when I use it in the animation part:

```
Argument of type 'ComponentFixture<TestAnimationComponent>' is not assignable to parameter of type 'ComponentFixture<TestComponent>'
```

The possible fixes I found were:

- Remove the type in the signature of getToastElement:
`getToastElement(fixture) ...`, but we lost the type feature in this function.
- Extract TestAnimationComponent from its `it` context and move it to the top, then define getToastElement as follow : `const getToastElement = (fixture: ComponentFixture<TestComponent | TestAnimationComponent>)`

I ended up by follow what is done in alert.specs.ts.

What do you prefer ? Any other idea ?